src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -51,7 +51,7 @@ public class TxPool : ITxPool, IDisposable
 
         private readonly object _locker = new();
 
-        private readonly ConcurrentDictionary<Address, AddressNonces> _nonces =
+        private readonly Dictionary<Address, AddressNonces> _nonces =
```

Comment:
why change from concurrent to normal?
could we bring it back to Concurrent

Reply:
Its always accessed under lock(), I don't think we need to make it concurrent then - please review.